### ISSUE TITLE
MGFX: Implement HLSL language support for .fx files

### ISSUE THREAD

**Issue by seclerp** [seclerp/rider-monogame#8]
Currently .fx files are just plain text ones. Need to implement syntax for them

Grammar to start from: https://github.com/SickheadGames/HL2GLSL/blob/master/parser/hlsl.grammar

Grammar used in Chromium: https://chromium.googlesource.com/chromium/chromium/+/d12b0baace92618bc8516561a254c0e96cd3a408/o3d/compiler/hlsl/HLSL.g

**Comment by jwosty** [comment]
Seems like Rider has some language support for HLSL. Is there any way that could be leveraged, at least as a stopgap? https://www.jetbrains.com/help/rider/Unreal_Engine__HLSL_Shaders.html

Main thing missing is the `technique` syntax, but I'd rather have any syntax highlighting at all.

**Comment by jwosty** [comment]
How involved is this type of change (for someone who has never worked on an JetBrains plugin)? If it's the kind of thing that can be done in an afternoon I'd be happy to give it a crack, if you point me in the right direction (i.e. what files need to be added and where, how do I build and test it locally, etc).

**Comment by seclerp** [comment]
> Seems like Rider has some language support for HLSL

You're totally right and this is the way I want to implement it once I have enough time for it :) We have a comprehensive HLSL language infrastructure built on top of powerful C++ infrastructure which may be reused.

> How involved is this type of change (for someone who has never worked on an JetBrains plugin)?

Unfortunately, by design our HLSL language support is done in a monolith C++ tooling which is not pluggable from outside, which means it requires main work to be done with the Rider codebase, not much with the plugin itself.
